lib/msf/core/post/linux/kernel.rb

```diff
@@ -0,0 +1,69 @@
+# -*- coding: binary -*-
+require 'msf/core/post/common'
+
+module Msf
+class Post
+module Linux
+module Kernel
+  include ::Msf::Post::Common
+
+  #
+  # Returns the kernel version
+  #
+  # @return [String]
+  #
+  def kernel_version
+    cmd_exec('uname -r').to_s
+  rescue
+    raise 'Could not determine kernel version'
+  end
+
+  #
+  # Returns the kernel modules
+  #
+  # @return [String]
```

Comment:
May be better to return an array of the modules rather than the string representation, which is filled with newlines and perhaps extra data that isn't necessary for common uses of this method.  

```
cmd_exec('cat /proc/modules').to_s.scan(/^[^ ]+/)
```

Reply:
I was thinking a `Hash`, but an `Array` would KISS.

Not sure how consistent the output is between Linuxes. I'll implement your change and we can burn that bridge when we get to it.

Reply:
Done